doc/reference/kernel/other/fatal.rst

```diff
@@ -0,0 +1,245 @@
+.. _fatal:
+
+Fatal Errors
+############
+
+Software Errors Triggered in Source Code
+****************************************
+
+Zephyr provides several methods for inducing fatal error conditions through
+either build-time checks, conditionally compiled assertions, or deliberately
+invoked panic or oops conditions.
+
+Runtime Assertions
+==================
+
+Zephyr provides some macros to perform runtime assertions which may be
+conditionally compiled. Their defintions may be found in
+``include/sys/__assert.h``.
+
+Assertions are enabled by setting the ``__ASSERT_ON`` preprocessor symbol to a
+non-zero value. There are two ways to do this:
+
+- Use the ``CONFIG_ASSERT`` and ``CONFIG_ASSERT_LEVEL`` kconfig options.
```

Comment:
This will create links to the config documentation for these options:
```
- Use the :option:`CONFIG_ASSERT` and :option`CONFIG_ASSERT_LEVEL`
  kconfig options.
```